src/Resharper.ConfigurationSense/Extensions/ProjectExtensions.cs

```diff
@@ -81,11 +82,16 @@ public static class ProjectExtensions
                         }
                     }
 
-                    settings.Add(new KeyValueSetting(formattedPath, property.Value.ToString()));
+                    if (!settings.ContainsKey(formattedPath))
```

Comment:
We can optimize this to access the dictionary only once with the `TryGet` method